src/utils/JSONUtils.ts

```diff
@@ -83,3 +83,37 @@ export function jsonReviverWithBigNumbers(_key: string, value: unknown): unknown
   }
   return value;
 }
+
+/**
+ * A replacer for use in `JSON.stringify` that converts BigInt number objects containing numeric string value.
+ * @param _key Unused
+ * @param value The value to convert
+ * @returns The converted object
+ */
+export function jsonReplacerWithBigInts(_key: string, value: unknown): unknown {
+  if (typeof value === "bigint") {
+    return { type: "BigInt", value: value.toString() };
```

Comment:
Alternative would be to encode this as decimal strings followed by `n` character which would take bit less space.

Reply:
I favour storing them as strings :+1:. Ultimately I wonder if we'll just use BigInt everywhere, in which case we could unconditionally convert strings as numbers back to BigInt.